src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -0,0 +1,224 @@
+
+    public ChannelService(ConversationParticipantRepository conversationParticipantRepository, ChannelRepository channelRepository, UserRepository userRepository,
+            ConversationService conversationService) {
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.channelRepository = channelRepository;
+        this.userRepository = userRepository;
+        this.conversationService = conversationService;
+    }
+
+    /**
+     * Gets the channel with the given id or throws a not found exception
+     *
+     * @param channelId the id of the channel
+     * @return the channel with the given id
+     */
+    public Channel getChannelOrThrow(Long channelId) {
+        return channelRepository.findById(channelId)
+                .orElseThrow(() -> new BadRequestAlertException("Channel with id " + channelId + " does not exist", CHANNEL_ENTITY_NAME, "idnotfound"));
+    }
+
+    /**
+     * Grans the channel admin role to the given user for the given channel
+     *
+     * @param channel                  the channel
+     * @param usersToGrantChannelAdmin the users to grant channel admin
+     */
+    public void grantChannelAdmin(Channel channel, Set<User> usersToGrantChannelAdmin) {
+        var matchingParticipants = conversationParticipantRepository.findConversationParticipantsByConversationIdAndUserIds(channel.getId(),
+                usersToGrantChannelAdmin.stream().map(User::getId).collect(Collectors.toSet()));
+        for (ConversationParticipant conversationParticipant : matchingParticipants) {
+            conversationParticipant.setIsAdmin(true);
+        }
+        conversationParticipantRepository.saveAll(matchingParticipants);
+        conversationService.notifyAllConversationMembersAboutUpdate(channel);
+    }
+
+    /**
+     * Removes the channel admin role from a user for the given channel
+     *
+     * @param channel                   the channel to remove the channel admin role from
+     * @param usersToRevokeChannelAdmin the users to revoke channel admin
+     */
+    public void revokeChannelAdmin(Channel channel, Set<User> usersToRevokeChannelAdmin) {
+        var matchingParticipants = conversationParticipantRepository.findConversationParticipantsByConversationIdAndUserIds(channel.getId(),
+                usersToRevokeChannelAdmin.stream().map(User::getId).collect(Collectors.toSet()));
+        for (ConversationParticipant conversationParticipant : matchingParticipants) {
+            conversationParticipant.setIsAdmin(false);
+        }
+        conversationParticipantRepository.saveAll(matchingParticipants);
+        conversationService.notifyAllConversationMembersAboutUpdate(channel);
+    }
+
+    /**
+     * Get all channels for the given course
+     *
+     * @param courseId the id of the course
+     * @return a list of channels for the given course
+     */
+    public List<Channel> getChannels(Long courseId) {
```

Comment:
Call the repository method directly?

Reply:
I do not like to break the Resource -> Service -> Repository layer that often tbh. I think its not "bad" to have it in the service. So I do not need to add a new dependency to the resource